isb_web/isb_solr_query.py

```diff
@@ -0,0 +1,221 @@
+    # that has a count > 0 and set the "count" property of the
+    # feature to that value.
+    for i_row in range(0, hm["rows"]):
+        for i_col in range(0, hm["columns"]):
+            if hm["counts_ints2D"][i_row] is not None:
+                v = hm["counts_ints2D"][i_row][i_col]
+                if v > 0:
+                    if v > _max_value:
+                        _max_value = v
+                    p0lat = lat_0 - dd_lat * i_row
+                    p0lon = lon_0 + dd_lon * i_col
+                    pts = geojson.Polygon(
+                        [
+                            [
+                                (
+                                    p0lon,
+                                    p0lat,
+                                ),
+                                (
+                                    p0lon + dd_lon,
+                                    p0lat,
+                                ),
+                                (
+                                    p0lon + dd_lon,
+                                    p0lat - dd_lat,
+                                ),
+                                (
+                                    p0lon,
+                                    p0lat - dd_lat,
+                                ),
+                                (
+                                    p0lon,
+                                    p0lat,
+                                ),
+                            ]
+                        ]
+                    )
+                    feature = geojson.Feature(geometry=pts, properties={"count": v})
+                    grid.append(feature)
+    grid.append(geojson.Feature(properties={"max_value": _max_value}))
+    grid.append(geojson.Feature(properties={"grid_level": gl}))
+    return geojson.FeatureCollection(grid)
+
+
+# Generate a list of [latitude, longitude, value] from
+# a solr heatmap. Latitude and longitude represent the
+# centers of the solr heatmap grid cells. The value is the count
+# for the grid cell.
+def solr_leaflet_heatmap(q, bb, grid_level=None):
+    hm = _get_heatmap(q, bb, _LEAFLET_ERR_PCT, grid_level)
+    # logging.warning(hm)
+    d_lat = hm["maxY"] - hm["minY"]
+    dd_lat = d_lat / (hm["rows"])
+    d_lon = hm["maxX"] - hm["minX"]
+    dd_lon = d_lon / (hm["columns"])
+    lat_0 = hm["maxY"] - dd_lat / 2.0
+    lon_0 = hm["minX"] + dd_lon / 2.0
+    data = []
+    max_value = 0
+    for i_row in range(0, hm["rows"]):
+        for i_col in range(0, hm["columns"]):
+            if hm["counts_ints2D"][i_row] is not None:
+                v = hm["counts_ints2D"][i_row][i_col]
+                if v > 0:
+                    lt = lat_0 - dd_lat * i_row
+                    lg = lon_0 + dd_lon * i_col
+                    data.append([lt, lg, v])
+                    if v > max_value:
+                        max_value = v
+    # return list of [lat, lon, count] and maximum count value
+    return {"data": data, "max_value": max_value}
```

Comment:
I ended up just putting this in a common dictionary as it seemed like a better fit than the previous Tuple.  My guess is we'll just need to adjust the callers to use this, but we were going to need to do that anyway.  What do you think?